File: alert-dashboard-service/src/main/java/com/aisiem/alert/domain/SecurityEvent.java
package com.aisiem.alert.domain;

import jakarta.persistence.*;
import lombok.*;

import java.time.LocalDateTime;

@Entity
@Table(name = "security_event")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class SecurityEvent {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "log_entry_id")
    private String logEntryId;

    @Column(name = "event_type", nullable = false, length = 50)
    private String eventType;

    @Column(nullable = false, length = 20)
    private String severity;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "source_ip", length = 45)
    private String sourceIp;

    @Column(name = "detected_by", nullable = false, length = 20)
    private String detectedBy;

    @Column(name = "rule_id")
    private Long ruleId;

    private Double confidence;

    @Column(length = 30)
    private String status;

    @Column(name = "raw_log", columnDefinition = "TEXT")
    private String rawLog;

    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}

File: threat-detection-service/app/model/security_event.py
from sqlalchemy import Column, BigInteger, String, Double, Text, TIMESTAMP, ForeignKey, func
from sqlalchemy.orm import declarative_base

Base = declarative_base()


class DetectionRule(Base):
    __tablename__ = "detection_rule"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    name = Column(String(255), nullable=False)
    description = Column(Text)
    event_type = Column(String(50), nullable=False)
    pattern = Column(Text, nullable=False)
    severity = Column(String(20), nullable=False)
    enabled = Column(String(5), default="1")
    created_at = Column(TIMESTAMP, server_default=func.now())
    updated_at = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now())


class SecurityEvent(Base):
    __tablename__ = "security_event"

    id = Column(BigInteger, primary_key=True, autoincrement=True)
    log_entry_id = Column(String(255))
    event_type = Column(String(50), nullable=False)
    severity = Column(String(20), nullable=False)
    description = Column(Text)
    source_ip = Column(String(45))
    detected_by = Column(String(20), nullable=False)  # RULE or AI
    rule_id = Column(BigInteger, ForeignKey("detection_rule.id", ondelete="SET NULL"))
    confidence = Column(Double, default=0.0)
    status = Column(String(30), default="NEW")
    raw_log = Column(Text)
    created_at = Column(TIMESTAMP, server_default=func.now())
    updated_at = Column(TIMESTAMP, server_default=func.now(), onupdate=func.now())
